microsoft/testsuites/nested/common.py

```diff
@@ -94,6 +95,131 @@ def connect_nested_vm(
     return nested_vm
 
 
+def hyperv_connect_nested_vm(
+    host: RemoteNode,
+    guest_username: str,
+    guest_password: str,
+    port: int,
+    guest_image_url: str,
+    name: str = "L2-VM",
+    image_name: str = HYPERV_NESTED_VM_IMAGE_NAME,
+    switch_name: str = "nestedvmswitch",
+    nat_name: str = "nestedvmnat",
+) -> RemoteNode:
+    # Download nested vm image
+    image_folder_path = "C:\\Users\\lisatest"
+    file_path = f"{image_folder_path}\\{image_name}"
+    host.tools[WindowsFileManager].download_file(guest_image_url, file_path)
+
+    # setup NAT
+    host.tools[HyperV].setup_nat_networking(switch_name, nat_name)
+    host.tools[HyperV].create_vm(
+        name,
+        file_path,
+        switch_name,
+    )
+
+    # enable port forwarding
+    local_ip = host.tools[HyperV].get_ip_address(name)
+    host.tools[HyperV].setup_port_forwarding(nat_name, port, local_ip)
+
+    # setup connection to nested vm
+    nested_vm = RemoteNode(Node(name=name), 0, name)
+    nested_vm.set_connection_info(
+        public_address=host.public_address,
+        username=guest_username,
+        password=guest_password,
+        public_port=port,
+        port=port,
+    )
+
+    # wait for nested vm ssh connection to be ready
+    try_connect(
+        ConnectionInfo(
```

Comment:
You can create a connection info object. It can be used field_to_dict to convert it to dict, and use it when `set_connection_info`

Reply:
I didn't quite get this comment. We use information from different places to create this `ConnectionInfo` object.

Reply:
The Ln 141 uses the variables as well. I mean, you can use the same connection info there by field_to_dict method. It saves lines of code.

Reply:
I am not sure if we can use it directly. `set_connection_info` requires a field called `public_port` which is not present in `ConnectionInfo` object. So, directly applying `fields_to_dict` would not work. We will need to anyways set it explicitly in dictionary before passing which I feel is making code more complicated

Reply:
It's ok, there are default value. Either port and public_port is set, it will be used by another one.

Reply:
Modified in latest commit